src/main/java/com/example/springbootrestexample/logic/BookLogic.java

```diff
@@ -16,8 +18,10 @@
 public class BookLogic extends ExceptionMessageLogic {
 
     @Autowired
+    @Qualifier("bookRepository")
```

Comment:
Remove qualifiers

Reply:
and autowires

Reply:
Dependencies should be added in the constructor, so the colaborators should have the modifiers
private final...